tests/conftest.py

```diff
@@ -42,3 +45,62 @@ def clean_hf_folder_token_for_tests() -> Generator:
     # Set back token once all tests have passed
     if token is not None:
         HfFolder().save_token(token)
+
+
+@pytest.fixture
+def fx_production_space(request: SubRequest) -> Generator[None, None, None]:
+    """Add a `repo_id` attribute referencing a space repo on the production Hub.
+
+    Fully testing Spaces is not possible on staging to we need to use the production
+    environment for it. Tests are skipped if we can't find a `HUGGINGFACE_PRODUCTION_USER_TOKEN`
+    environment variable.
+
+    Example:
+    ```py
+    @pytest.mark.usefixtures("fx_production_space")
+    class TestSpaceAPI(unittest.TestCase):
+        repo_id: str
+        api: HfApi
+
+        def test_space(self) -> None:
+            api.repo_info(repo_id, repo_type="space")
+    ```
+    """
+    # Check if production token exists
+    if PRODUCTION_TOKEN is None:
+        pytest.skip(
+            "Skip Space tests. `HUGGINGFACE_PRODUCTION_USER_TOKEN` environment variable"
+            " is not set."
+        )
+
+    # Generate repo id from prod token
+    api = HfApi(token=PRODUCTION_TOKEN, endpoint=ENDPOINT_PRODUCTION)
+    user = api.whoami()["name"]
+    repo_id = f"{user}/{repo_name(prefix='tmp_test_space')}"
+    request.cls.api = api
+    request.cls.repo_id = repo_id
+
+    # Create and clean space repo
+    api.create_repo(
+        repo_id=repo_id, repo_type="space", space_sdk="gradio", private=True
+    )
+    api.upload_file(
+        path_or_fileobj=_BASIC_APP_PY_TEMPLATE,
+        repo_id=repo_id,
+        repo_type="space",
+        path_in_repo="app.py",
+    )
+    yield
```

Comment:
@osanseviero This will execute the code above the `yield` before the test and the code after the `yield` after it. It is equivalent to the `setUp` / `tearDown` syntax but in a single function.

It is very convenient (-at least I find it convenient-) to separate code when you have a lot of fixtures and you want to be sure that you don't forget to undo something.
```py
# grouped
def setUp():
    doA()
    doB()

def tearDown():
    undoA()
    undoB()
    
# independent
def fixtureA():
    doA()
    yield
    undoA()

def fixtureB():
    doB()
    yield
    undoB()
```


See [pytest docs](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended).